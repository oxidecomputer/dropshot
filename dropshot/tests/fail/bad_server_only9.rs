// Copyright 2024 Oxide Computer Company

#![allow(unused_imports)]

// Test for a trait with a lifetime parameter.

#[dropshot::server]
trait MyServer<'a> {
    type Context;
}

enum MyImpl {}

// This should not produce errors about the trait or the context type being
// missing.
impl<'a> MyServer<'a> for MyImpl {
    type Context = ();
}

fn main() {
    // These items will NOT be present because of the invalid trait, and will
    // cause errors to be generated.
    my_server::api_description::<MyImpl>();
    my_server::stub_api_description();
}
