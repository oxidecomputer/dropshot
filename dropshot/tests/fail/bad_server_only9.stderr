error: server `MyServer` must not have generics
 --> tests/fail/bad_server_only9.rs:8:15
  |
8 | trait MyServer<'a> {
  |               ^^^^

error[E0425]: cannot find function `api_description` in module `my_server`
  --> tests/fail/bad_server_only9.rs:23:16
   |
23 |     my_server::api_description::<MyImpl>();
   |                ^^^^^^^^^^^^^^^ not found in `my_server`

error[E0425]: cannot find function `stub_api_description` in module `my_server`
  --> tests/fail/bad_server_only9.rs:24:16
   |
24 |     my_server::stub_api_description();
   |                ^^^^^^^^^^^^^^^^^^^^ not found in `my_server`
